src/libopensc/card-openpgp.c

```diff
@@ -1188,20 +1234,29 @@ pgp_enumerate_blob(sc_card_t *card, pgp_blob_t *blob)
 
 		r = sc_asn1_read_tag(&data, blob->len - (in - blob->data),
 					&cla, &tag, &len);
-		if (r < 0 || data == NULL) {
-			sc_log(card->ctx,
-				 "Unexpected end of contents\n");
+		if (data == NULL) {
```

Comment:
it is safer to keep both checks here -- some static analyzers might get confused by checking only one of these.

Reply:
I've split this check into three - `data = NULL` is in the same place, then `r == SC_ERROR_INVALID_ASN1_OBJECT`, to have a better message, than "Unexpected end of contents", and `r == SC_ERROR_ASN1_END_OF_CONTENTS`, to handle Yubikey issue. `sc_asn1_read_tag` can return only SC_ERROR_INVALID_ASN1_OBJECT and SC_ERROR_ASN1_END_OF_CONTENTS, so, I think, I covered all cases?

Reply:
Or do you mean, that static analyzers are not aware, that there are only two possible errors in called function, and want that calling function should check for r<0 anyway?

Reply:
Sorry, I missed that part that you moved the `r` checking to the other places. Right now, it returns these two errors, but it might change in the future and we might get different return values which would go unnoticed. So I would be for adding a fallback branch for all the `r < 0` anyway.

Reply:
Got it, will add, thank you!